### PR TITLE
Fixes #8229: Add constraint in ncf technique editor

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/8229

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
Commit modified

File: builder/js/ncf.js [hunk 1]
```diff
@@ -46,232 +46,279 @@ app.directive('techniquename', function($filter) {
...
+$scope.getMethodsAndTechniques = function () {
+  var data = {params: {path: $scope.path}}
+  $http.get('/ncf/api/generic_methods', data).
+    success(function(data, status, headers, config) {
+      $scope.generic_methods = data.data.generic_methods;
+      $scope.constraints = data.data.constraints;
+      $scope.methodsByCategory = $scope.groupMethodsByCategory();
+      $scope.authenticated = true;
+      // Once we have our methods we can fetch our techniques which depends on them
+      $scope.getTechniques();
```
Comment (fanf) [inline builder/js/ncf.js]: This fails if not techniques is defined (must data.data.techniques !== undefined in gettechniques, or something alike)

**Comment by fanf** [comment]
This seems to work as expected, and expect for the comments for undefined techniques, there is one thing: error message, for ex on a select, are displayed: [u'==', u'<=', u'>=', u'<', u'>', u'!='] in place of ['==', ... => please remove the "u" for python utf-8 string.

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by fanf** [comment]
u'==' etc are still there :/

**Comment by fanf** [comment]
I'm merging it, because we are not going to block that feature for one more week for that, but please open a ticket for the u''

**Pull request merged**